.github/workflows/maven.yaml

```diff
@@ -0,0 +1,38 @@
+name: Java CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: '*'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # required by wagoid/commitlint-github-action to validate range of commits instead of just the latest one
+      - uses: wagoid/commitlint-github-action@v6
+        if: matrix.os == 'ubuntu-latest'
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v4
+        with:
+          distribution: corretto
+          java-version: 8
+          cache: maven
+      - run: rm -rf /tmp/*
+        if: matrix.os == 'ubuntu-latest'
+        continue-on-error: true
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          version: v0.4.0
```

Comment:
what's this for?

Reply:
Without this, latest version of docker buildx is used and that doesn't support `.` in the docker bake file used by the parsec client library. We could cut a PR to fix that but I didn't want to wait on that. So, I've updated the CI to use the version of docker buildx that supports the format. 

https://github.com/parallaxsecond/parsec-client-java/blob/1e44b25e63fdf2aa430c33bdc796afd5c61d5da7/parsec-testcontainers/docker-bake.hcl#L28

```ERROR: docker-bake.hcl:28,8-22: Invalid name; only "[a-zA-Z0-9_-]+" are allowed``` as seen here https://github.com/awslabs/aws-greengrass-labs-parsec-provider/actions/runs/8710795502/job/23893455737#step:7:99

Reply:
ok